egs/imagenet/v1/local/process_data.py

```diff
@@ -0,0 +1,181 @@
+    crop_topleft_fliplr = np.fliplr(crop_topleft)
+    crop_topright = im[0:crop_size,(W-crop_size):W]
+    crop_topright_fliplr = np.fliplr(crop_topright)
+    crop_bottomleft = im[(H-crop_size):H,0:crop_size]
+    crop_bottomleft_fliplr = np.fliplr(crop_bottomleft)
+    crop_bottomright = im[(H-crop_size):H,(W-crop_size):W]
+    crop_bottomright_fliplr = np.fliplr(crop_bottomright)
+
+    center_left = float(W)/2 - float(crop_size)/2
+    center_top = float(H)/2 - float(crop_size)/2
+    crop_center = im[center_top:(center_top+crop_size),
+                     center_left:(center_left+crop_size)]
+    crop_center_fliplr = np.fliplr(crop_center)
+
+    tup1 = (crop_topleft, crop_topright,
+            crop_bottomleft, crop_bottomright,
+            crop_center)
+    tup2 = (crop_topleft_fliplr, crop_topright_fliplr,
+            crop_bottomleft_fliplr, crop_bottomright_fliplr,
+            crop_center_fliplr)
+    tup = tup1 + tup2
+    return tup
+
+### main ###
+mat_path, val_ground_truth_path = parse_mat_path()
+mat_content = sio.loadmat(mat_path)
+synsets_struct = mat_content['synsets']
+wnid_vect = synsets_struct['WNID']
+
+if args.out_ark == '-':
+    out_fh = sys.stdout
+else:
+    out_fh = open(args.out_ark,'wb')
+
+labels_file = os.path.join(args.dir + '/', 'labels.txt')
+labels_fh = open(labels_file,'wb')
+
+database_contents = sorted(os.listdir(args.databasePath))
+
+if args.dataset == 'train':
+    image_id = 1
+    for dir_file in database_contents:
+        potential_path = args.databasePath + '/' + dir_file
+        if os.path.isdir(potential_path):
+            index = findIndex(wnid_vect,dir_file)
+            class_ID = synsets_struct[index][0][0][0][0]
+            class_contents = sorted(os.listdir(potential_path))
+            for img_Name in class_contents:
+                key = zeropad(image_id,8)
+                im = misc.imread(potential_path + '/' + img_Name)
+		im = np.divide(im,255.0)
+                W = im.shape[1]
+                H = im.shape[0]
+                if len(im.shape) == 3:
+                    data = np.reshape(np.transpose(im, (1, 0, 2)), (W, H * C))
+                else:
+                    im_three = np.dstack((im, im, im))
+                    data = np.reshape(np.transpose(im_three, (1, 0, 2)), (W, H * C))
+                labels_fh.write(key + ' ' + str(int(class_ID)-1) + '\n')
+                write_kaldi_matrix(out_fh, data, key)
+                image_id = image_id + 1
+else:
+    image_id = 1
+    image_num = 1
+    datasetYearVect = args.tarName.split('_')
+    with open(val_ground_truth_path) as f:
+        for line in f:
+            if int(line) > 0:
+                keyID = zeropad(image_id,8)
+                file_name = (args.databasePath + '/'
+                            + datasetYearVect[0] + '_val_' + keyID + '.JPEG')
+                im_orig = misc.imread(file_name)
+		im_orig = np.divide(im_orig,255.0)
+		im_tup = cropImage(im_orig)
```

Comment:
It's better to use more clear variable/function names. For e.g. `image_crops` instead of `im_tup` and `get_image_crops()` instead of `cropImage()`.
Also, there are still a few tabs in this file.